source/game_sa/3dMarkers.cpp

```diff
@@ -1,45 +1,404 @@
+    return plugin::CallAndReturn<C3dMarker*, 0x725120, uint32, e3dMarkerType, CVector&, float, uint8, uint8, uint8, uint8, uint16, float, int16, float, float, float, bool>(id, type, posn, size, red, green, blue, alpha, pulsePeriod, pulseFraction, rotateRate, nrm_x, nrm_y, nrm_z, zCheck);
+}
+
+// 0x726D40
+void C3dMarkers::PlaceMarkerCone(uint32 id, CVector& point, float size, uint8 red, uint8 green, uint8 blue, uint8 alpha, uint16 pulsePeriod, float pulseFraction, int16 rotateRate, bool bEnableCollision) {
+    if ((point - TheCamera.GetPosition()).SquaredMagnitude() >= sq(1.6f)) {
+        PlaceMarker(id, bEnableCollision ? MARKER3D_CONE : MARKER3D_CONE_NO_COLLISION, point, size, red, green, blue, m_colDiamond, pulsePeriod, pulseFraction, 0, 0.0f, 0.0f, 0.0f, false);
+    }
+}
+
+// 0x725BA0
+void C3dMarkers::PlaceMarkerSet(uint32 id, e3dMarkerType type, CVector& posn, float size, uint8 red, uint8 green, uint8 blue, uint8 alpha, uint16 pulsePeriod, float pulseFraction, int16 rotateRate) {
+    PlaceMarker(id, type, posn, size, red, green, blue, static_cast<uint8>((float)alpha * 1.0f / 3.0f), pulsePeriod, pulseFraction, 1, 0.0f, 0.0f, 0.0f, false);
+}
+
+// 0x7210D0
+// only set material color (m_user3dMarkerColor) for first material in first atomic; 'data' is unused
+RpAtomic* C3dMarkers::User3dMarkerAtomicCB(RpAtomic* atomic, void*) {
+    const auto color = m_user3dMarkerColor.ToRwRGBA();
+    RpMaterialSetColor(RpGeometryGetMaterial(RpAtomicGetGeometry(atomic), 0), &color);
+    return nullptr;
+}
+
+// 0x721090
+void C3dMarkers::User3dMarkerDelete(int32 slotIndex) {
+    if (slotIndex >= 0 && slotIndex <= (int32)(ms_user3dMarkers.size() - 1)) {
+        ms_user3dMarkers[slotIndex].m_bIsUsed = false;
+    }
+}
+
+// 0x7210B0
+void C3dMarkers::User3dMarkerDeleteAll() {
+    for (auto& marker : ms_user3dMarkers) {
+        marker.m_bIsUsed = false;
+    }
+}
+
+// Returns slot index; -1 if not found
+// 0x720FB0
+int32 C3dMarkers::User3dMarkerFindFirstFreeSlot() {
+    for (const auto& [index, marker] : notsa::enumerate(ms_user3dMarkers)) {
+        if (!marker.m_bIsUsed) {
+            return index;
+        }
+    }
+    return -1;
+}
+
+// Returns slot index; -1 if not created; for 'color', see eHudColours
+// 0x720FD0
+int32 C3dMarkers::User3dMarkerSet(float x, float y, float z, eHudColours color) {
+    const auto markerIndex = User3dMarkerFindFirstFreeSlot();
+    if (markerIndex == -1) {
+        return -1;
+    }
+
+    auto& marker = ms_user3dMarkers[markerIndex];
+
+    marker.m_vecPosition.Set(x, y, z);
+
+    const auto colour = HudColour.GetRGB(color);
+    marker.m_nRed = colour.r;
+    marker.m_nGreen = colour.g;
+    marker.m_nBlue = colour.b;
+
+    marker.m_bIsUsed = true;
+
+    return markerIndex;
```

Comment:
```suggestion
    if (markerIndex != -1) {
        auto& marker = ms_user3dMarkers[markerIndex];

        marker.m_vecPosition.Set(x, y, z);
    
        const auto colour = HudColour.GetRGB(color);
        marker.m_nRed = colour.r;
        marker.m_nGreen = colour.g;
        marker.m_nBlue = colour.b;

        marker.m_bIsUsed = true;
    }

    return markerIndex;
```
one level of indent does not hurt imo

Reply:
Okay, fair.